Core 20 highlight edit navigation

RoyEJohnson opened the pull request:
[CORE-20]
[CORE-459]
The latter one describes using tab to get to the dialog, but we took a different approach (we want to be able to tab through the highlights without the detours, and use a different key to go into the editor).

[CORE-20]: https://openstax.atlassian.net/browse/CORE-20?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CORE-459]: https://openstax.atlassian.net/browse/CORE-459?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

RoyEJohnson committed:
refactor CardWrapper and other code cleanup

RoyEJohnson committed:
Double-click selection to go to edit window

RoyEJohnson committed:
Edit on enter

RoyEJohnson committed:
EditCard only appears when focused

Update snapshots

RoyEJohnson committed:
Test coverage

jivey reviewed (commented):
Changes seem good, but after using it, I'm worried the hidden highlight popup won't be discoverable enough. But we can get some feedback using the deployed env.

RoyEJohnson committed:
Fix test?

RoyEJohnson committed:
Show EditCard with instructions for focused highlight without note

Fix coverage

RoyEJohnson commented:
@jivey I have modified the highlight editor to show instructions (only) when a highlight is focused, to make the editor more discoverable. The tests are not fixed yet, but you can see whether you think it's better.

jivey reviewed (approved):
I see the help popup when unauthenticated, it seems a bit narrow though:
<img width="108" height="157" alt="image" src="https://github.com/user-attachments/assets/9ff87482-bad4-4ec1-a5b6-391c62a620d0" />

Otherwise I think this is good, and the keyboard interactions make a lot of sense.

RoyEJohnson committed:
Show instructions for authenticated/not

If not authenticated, show login button
If no annotation, show instruction box
If annotation or edit-highight sequence has happened, show edit card

RoyEJohnson committed:
Update jira-linked-action

staxly[bot] committed:
Merge branch 'main' into core-20-highlight-edit-navigation

staxly[bot] committed:
Merge branch 'main' into core-20-highlight-edit-navigation
